cli/src/main.rs

```diff
@@ -0,0 +1,698 @@
+
+    /// Interactive Perspective shell based on Prolog/SDNA runtime
+    Repl { id: String },
+}
+
+#[derive(Args, Debug)]
+struct QueryLinksArgs {
+    /// Perspective ID
+    id: String,
+
+    /// Filter by source
+    source: Option<String>,
+
+    /// Filter by target
+    target: Option<String>,
+
+    /// Filter by predicate
+    predicate: Option<String>,
+
+    /// Get only links after this date (fromat: %Y-%m-%dT%H:%M:%S%.fZ)
+    #[arg(short, long)]
+    from_date: Option<String>,
+
+    /// Get only links before this date (fromat: %Y-%m-%dT%H:%M:%S%.fZ)
+    #[arg(short, long)]
+    until_date: Option<String>,
+
+    /// Get only the first n links
+    #[arg(short, long)]
+    limit: Option<f64>,
+}
+
+#[derive(Debug, Subcommand)]
+enum NeighbourhoodFunctions {
+    Create {
+        perspective_id: String,
+        link_language: String,
+    },
+    Join {
+        url: String,
+    },
+}
+
+#[derive(Debug, Subcommand)]
+enum RuntimeFunctions {
+    Info,
+    Quit,
+    AddTrustedAgents {
+        agents: Vec<String>,
+    },
+    DeleteTrustedAgents {
+        agents: Vec<String>,
+    },
+    TrustedAgents,
+    AddLinkLanguageTemplates {
+        addresses: Vec<String>,
+    },
+    RemoveLinkLanguageTemplates {
+        addresses: Vec<String>,
+    },
+    LinkLanguageTemplates,
+    AddFriends {
+        agents: Vec<String>,
+    },
+    RemoveFriends {
+        agents: Vec<String>,
+    },
+    Friends,
+    HcAgentInfos,
+    HcAddAgentInfos {
+        infos: String,
+    },
+    VerifySignature {
+        did: String,
+        did_signing_key: String,
+        data: String,
+        signed_data: String,
+    },
+    SetStatus {
+        status: String,
+    },
+    FriendStatus {
+        agent: String,
+    },
+    FriendSendMessage {
+        agent: String,
+        message: String,
+    },
+    MessageInbox {
+        filter: Option<String>,
+    },
+    MessageOutbox {
+        filter: Option<String>,
+    },
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let args = ClapApp::parse();
+    let cap_token = match &args.domain {
+        Domain::Log => "".to_string(),
+        Domain::Agent { command } => match command {
+            AgentFunctions::Lock | AgentFunctions::Unlock => "".to_string(),
+            _ => startup::get_cap_token().await?,
+        },
+        _ => startup::get_cap_token().await?,
+    };
+
+    match args.domain {
+        Domain::Agent { command } => match command {
+            AgentFunctions::Me => {
+                let agent = agent::run_me(cap_token).await?;
+                print_agent(agent.into());
+            }
+            AgentFunctions::Status => {
+                let status = agent::run_status(cap_token).await?;
+                println!(
+                    "\x1b[36mDID: \x1b[97m{}",
+                    status.did.unwrap_or_else(|| "<undefined>".to_string())
+                );
+                println!("\x1b[36mis_initiliazed: \x1b[97m{}", status.is_initialized);
```

Comment:
Shall we put this whole match block inside its own function so we can easily call it from tests?

Reply:
I have extracted all functions into modules now.